microgateway_decorator/bin/compile

```diff
@@ -7,12 +7,95 @@ BUILD_DIR=$1
+{
+  echo "untar"
+  NODEJS_FILENAME=$1
+  tar --touch --strip-components 1 -xzf $NODEJS_FILENAME -C $BUILD_DIR/tmp/node
+}
+
+function wget_and_untar
+{
+  NODE_WGET=$1
+  NODEJS_FILENAME=$2
+  wget_output=$(wget --tries=1 --timeout=5 --quiet "$NODE_WGET")
+  if [ $? -ne 0 ]; then
+    echo "Node.js was not downloaded successfully from $NODE_WGET." >&2
+    exit 1
+  fi
+  untar $NODEJS_FILENAME
+  rm $NODEJS_FILENAME
+}
+
+# checks various combinations of the filename and the url and displays an error message
+# if the URL is valid protocol (http:// or https://)
+# or the url is missing and the file name is present
+# or the url is present and the file name is missing
+function check_nodejs_filename_and_url {
+  echo "Checking APIGEE_MICROGATEWAY_NODEJS_FILENAME and APIGEE_MICROGATEWAY_NODEJS_URL environment variables."
+  # check if the value is set to false or if it is zero length then assume false.
+  if [ "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" == false ] || [ -z "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" ]; then
+    echo "APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL is either set to false or assumed false."
+    if [ "$APIGEE_MICROGATEWAY_NODEJS_URL" != "" ] && [ "$APIGEE_MICROGATEWAY_NODEJS_FILENAME" == "" ]; then
+      echo "If you set APIGEE_MICROGATEWAY_NODEJS_URL=$APIGEE_MICROGATEWAY_NODEJS_URL then you must also set APIGEE_MICROGATEWAY_NODEJS_FILENAME as well." >&2
+      exit 1
+    elif [ "$APIGEE_MICROGATEWAY_NODEJS_URL" == "" ] && [ "$APIGEE_MICROGATEWAY_NODEJS_FILENAME" != "" ]; then
+      echo "If you set APIGEE_MICROGATEWAY_NODEJS_FILENAME=$APIGEE_MICROGATEWAY_NODEJS_FILENAME then you must also set APIGEE_MICROGATEWAY_NODEJS_URL as well." >&2
+      exit 1
+    else
+      if [ "$APIGEE_MICROGATEWAY_NODEJS_FILENAME" != "" ]; then
+        if ! grep -q "http://" <<< "$APIGEE_MICROGATEWAY_NODEJS_URL" && ! grep -q "https://" <<< "$APIGEE_MICROGATEWAY_NODEJS_URL" ; then
+          echo "$APIGEE_MICROGATEWAY_NODEJS_URL does not include a valid protocol (https or http)" >&2
+          exit 1
+        fi
+      fi
+    fi
+  elif [ $APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL == true ]; then
+    if [ ! -f "$ROOT_DIR/lib/$APIGEE_MICROGATEWAY_NODEJS_FILENAME" ]; then
+      echo "APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL is set to true but APIGEE_MICROGATEWAY_NODEJS_FILENAME=$APIGEE_MICROGATEWAY_NODEJS_FILENAME and that file does not exist in the lib directory. Exiting..." >&2
+      exit 1
+    fi
+  fi
+  echo "APIGEE_MICROGATEWAY_NODEJS_FILENAME and APIGEE_MICROGATEWAY_NODEJS_URL environment variables are set correctly."
+}
+
 mkdir -p $BUILD_DIR/tmp/node
 mkdir -p $BUILD_DIR/tmp/edgemicro
 
-wget --quiet https://nodejs.org/dist/v6.11.3/node-v6.11.3-linux-x64.tar.gz
-tar --touch --strip-components 1 -xzf node-v6.11.3-linux-x64.tar.gz -C $BUILD_DIR/tmp/node
-rm node-v6.11.3-linux-x64.tar.gz
+check_nodejslocalinstall_env_var
+check_nodejs_filename_and_url
+
+if [ "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" == true ]; then
```

Comment:
We should be able to just check if they have a local install file specified and can get rid of the double option here.

Reply:
Agreed. If the APIGEE_MICROGATEWAY_NODEJS_FILENAME var is included then it is a local install otherwise it is wget.